pkg/yang/modules.go

```diff
@@ -28,16 +28,27 @@ type Modules struct {
 	includes   map[*Module]bool   // Modules we have already done include on
 	byPrefix   map[string]*Module // Cache of prefix lookup
 	byNS       map[string]*Module // Cache of namespace lookup
+	typeDict   *typeDictionary    // Cache for type definitions.
 }
 
 // NewModules returns a newly created and initialized Modules.
 func NewModules() *Modules {
+	d := &typeDictionary{dict: map[Node]map[string]*Typedef{}}
+	// taking the lock should not be needed as this type should only be modified
+	// during init.
+	typeDict.mu.Lock()
+	defer typeDict.mu.Unlock()
+	for k, v := range typeDict.dict {
+		d.dict[k] = v
+	}
```

Comment:
I'm confused about this part -- wouldn't this be polluting the type space of the `Modules` object with whatever has been populated into the default/global `typeDict`, which I expect to be empty unless the user has decided to make lower-level calls (`Parse`, `BuildAST`)?